src/function/scalar/enum/enum_functions_implementation.cpp

```diff
@@ -17,6 +18,18 @@ static void EnumLastFunction(DataChunk &input, ExpressionState &state, Vector &r
 	result.Reference(val);
 }
 
+static void EnumRandomFunction(DataChunk &input, ExpressionState &state, Vector &result) {
+	D_ASSERT(input.GetTypes().size() == 1);
+	auto enum_size = EnumType::GetSize(input.GetTypes()[0]);
+	auto &enum_vector = EnumType::GetValuesInsertOrder(input.GetTypes()[0]);
+	std::random_device dev;
+	std::mt19937 rng(dev());
+	std::uniform_int_distribution<std::mt19937::result_type> dist(0, enum_size - 1);
+
+	auto val = Value(enum_vector.GetValue(dist(rng)));
```

Comment:
This generates one random value for the entire vector, rather than one random value per value, which seems unintended

Reply:
Ah, my understanding was one random value only, is it one per value then?

Reply:
You can do something like `SELECT enum_random(null::rainbow) FROM range(100)`, and I would expect 100 different random values to be generated

Reply:
fair enough